services/repository/src/main/java/org/sagebionetworks/repo/web/OAuthScopeInterceptor.java

```diff
@@ -61,10 +61,20 @@ public static boolean isAnonymous(HttpServletRequest request) {
 				AuthorizationConstants.BOOTSTRAP_PRINCIPAL.ANONYMOUS_USER.getPrincipalId()
 					.equals(Long.parseLong(userIdRequestParameter));
 	}
+	
+	public static boolean isServiceCall(HttpServletRequest request) {
+		String serviceName = request.getHeader(AuthorizationConstants.SYNAPSE_HEADER_SERVICE_NAME);
```

Comment:
need to make sure that a client cannot add this header to bypass the scope check